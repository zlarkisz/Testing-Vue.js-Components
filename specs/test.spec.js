import { mount } from "@vue/test-utils";

import TestComponent from "@/test";
import ListComponent from "@/list";

test("mount a vue component", () => {
  const wrapper = mount(TestComponent, {
    propsData: {
      value: "H@$ka",
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test("ListComponent", () => {
  const wrapper = mount(ListComponent);
  const movies = wrapper.vm.marvelMovies;
  wrapper.setData({ marvelMovies: [...movies, "Endgame"] });
  expect(wrapper).toMatchSnapshot();
});
