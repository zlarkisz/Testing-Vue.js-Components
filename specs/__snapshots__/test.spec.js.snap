// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
    <li><strong>Endgame</strong></li>
  </ul>
</div>
`;

exports[`mount a vue component 1`] = `
<div>
  <h1>Hello H@$ka</h1>
</div>
`;
